### PR TITLE
docs: improve `InstructionResult` documentation

### PR DESCRIPTION
**Pull request by leovct**
## Description

Minor updates to the `InstructionResult` docs.

Related to https://github.com/foundry-rs/foundry/issues/7463

### CODE REVIEW

**Review by rakita — COMMENTED** [review]
Will need to review those Enum's but lets merge comment

File: crates/interpreter/src/instruction_result.rs [hunk 1]
```diff
@@ -4,68 +4,91 @@ use crate::primitives::{HaltReason, OutOfGasError, SuccessReason};
...
-    /// Error on created contract that begins with EF
+    /// Created contract starts with invalid bytes (`0xEF`).
     CreateContractStartingWithEF,
-    /// EIP-3860: Limit and meter initcode. Initcode size limit exceeded.
+    /// Exceeded init code size limit (EIP-3860).
     CreateInitCodeSizeLimit,
     /// Fatal external error. Returned by database.
     FatalExternalError,
-    /// RETURNCONTRACT called in not init eof code.
+    /// `RETURNCONTRACT` called outside init Ethereum Object Format code.
```
Comment (rakita) [inline crates/interpreter/src/instruction_result.rs:81]: It is `EVM Object Format` but EOF is abbreviation is used
```suggestion
    /// `RETURNCONTRACT` called outside init EOF code.
```

File: crates/interpreter/src/instruction_result.rs [hunk 2]
```diff
@@ -4,68 +4,91 @@ use crate::primitives::{HaltReason, OutOfGasError, SuccessReason};
...
     ReturnContractInNotInitEOF,
-    /// Legacy contract is calling opcode that is enabled only in EOF.
+    /// Legacy contract is calling opcode that is enabled only in Ethereum Object Format.
```
Comment (rakita) [inline crates/interpreter/src/instruction_result.rs:83]:
```suggestion
    /// Legacy contract is calling opcode that is enabled only in EOF.
```

File: crates/interpreter/src/instruction_result.rs [hunk 3]
```diff
@@ -4,68 +4,91 @@ use crate::primitives::{HaltReason, OutOfGasError, SuccessReason};
...
     EOFOpcodeDisabledInLegacy,
-    /// EOF function stack overflow
+    /// Stack overflow in Ethereum Object Format function.
```
Comment (rakita) [inline crates/interpreter/src/instruction_result.rs:85]:
```suggestion
    /// Stack overflow in EOF subroutine function calls.
```

File: crates/interpreter/src/instruction_result.rs [hunk 4]
```diff
@@ -4,68 +4,91 @@ use crate::primitives::{HaltReason, OutOfGasError, SuccessReason};
...
+    /// Nonce overflow.
     NonceOverflow,
-    /// Create init code size exceeds limit (runtime).
+    /// Exceeded contract size limit during creation.
     CreateContractSizeLimit,
-    /// Error on created contract that begins with EF
+    /// Created contract starts with invalid bytes (`0xEF`).
     CreateContractStartingWithEF,
-    /// EIP-3860: Limit and meter initcode. Initcode size limit exceeded.
+    /// Exceeded init code size limit (EIP-3860).
```
Comment (rakita) [inline crates/interpreter/src/instruction_result.rs:77]:
```suggestion
    /// Exceeded init code size limit (EIP-3860:  Limit and meter initcode).
```

Comment (rakita) [reply crates/interpreter/src/instruction_result.rs:77]: `Limit and meter initcode` is a EIP name

File: crates/interpreter/src/instruction_result.rs [hunk 5]
```diff
@@ -4,68 +4,91 @@ use crate::primitives::{HaltReason, OutOfGasError, SuccessReason};
...
+    Revert = 0x10,
+    /// Exceeded maximum call depth.
     CallTooDeep,
+    /// Insufficient funds for transfer.
     OutOfFunds,
-    /// Revert if CREATE/CREATE2 starts with 0xEF00
+    /// Revert if `CREATE`/`CREATE2` starts with `0xEF00`.
     CreateInitCodeStartingEF00,
-    /// Invalid EOF initcode,
+    /// Invalid Ethereum Object Format (EOF) init code.
```
Comment (rakita) [inline crates/interpreter/src/instruction_result.rs:29]:
```suggestion
    /// Invalid EVM Object Format (EOF) init code.
```

File: crates/interpreter/src/instruction_result.rs [hunk 6]
```diff
@@ -4,68 +4,91 @@ use crate::primitives::{HaltReason, OutOfGasError, SuccessReason};
 #[derive(Clone, Copy, Debug, Default, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub enum InstructionResult {
-    // success codes
+    // Success Codes
     #[default]
+    /// Execution should continue to the next one.
     Continue = 0x00,
+    /// Halt execution.
```
Comment (rakita) [inline crates/interpreter/src/instruction_result.rs:11]: `Halt` can mean different behaviours
```suggestion
    /// Encountered a `STOP` opcode
```
Not sure how to phase this one

Comment (leovct) [reply crates/interpreter/src/instruction_result.rs:11]: I used the description shown in [evm.codes](https://www.evm.codes/#00?fork=cancun) but happy to change it.

Screenshot 2024-08-06 at 13 48 42

Comment (rakita) [reply crates/interpreter/src/instruction_result.rs:11]: Calling it halt is not correct, as STOP is considered valid/successful execution. Lets change it.

**Review by rakita — APPROVED** [review]
Thank you!

**Pull request merged**